apollo-router/src/plugins/authentication/subgraph_auth.rs

```diff
@@ -0,0 +1,455 @@
+                        Ok(credentials) => credentials,
+                        Err(err) => {
+                            tracing::warn!(
+                                "Failed to serialize GraphQL body for AWS SigV4 signing, skipping signing. Error: {}",
+                                err
+                            );
+                                return Ok(ControlFlow::Continue(req));
+                        }};
+                    let settings = get_signing_settings(&signing_params);
+                    let mut builder = http_request::SigningParams::builder()
+                        .access_key(credentials.access_key_id())
+                        .secret_key(credentials.secret_access_key())
+                        .region(signing_params.region.as_ref())
+                        .service_name(&signing_params.service_name)
+                        .time(SystemTime::now())
+                        .settings(settings);
+                    builder.set_security_token(credentials.session_token());
+                    let body_bytes = match serde_json::to_vec(&req.subgraph_request.body()) {
+                        Ok(b) => b,
+                        Err(err) => {
+                            tracing::warn!(
+                            "Failed to serialize GraphQL body for AWS SigV4 signing, skipping signing. Error: {}",
+                            err
+                        );
+                            return Ok(ControlFlow::Continue(req));
+                        }
+                    };
+                    // UnsignedPayload only applies to lattice
+                    let signable_request = SignableRequest::new(
+                        req.subgraph_request.method(),
+                        req.subgraph_request.uri(),
+                        req.subgraph_request.headers(),
+                        match signing_params.service_name.as_str() {
+                            "vpc-lattice-svcs" => SignableBody::UnsignedPayload,
+                            _ => SignableBody::Bytes(&body_bytes),
+                        },
+                    );
+
+                    let signing_params = builder.build().expect("all required fields set");
+
+                    let (signing_instructions, _signature) = match sign(signable_request, &signing_params) {
+                        Ok(output) => output,
+                        Err(err) => {
+                            tracing::warn!("Failed to sign GraphQL request for AWS SigV4, skipping signing. Error: {}", err);
+                            return Ok(ControlFlow::Continue(req));
+                        }
+                    }.into_parts();
+                    signing_instructions.apply_to_request(&mut req.subgraph_request);
+                     Ok(ControlFlow::Continue(req))
+                } else {
+                    Ok(ControlFlow::Continue(req))
+                }
+            }
+            }).buffered()
```

Comment:
can we get rid of buffered ?

Reply:
I think we can, once I turn the plugin into a layer to address https://github.com/apollographql/router/pull/3365#pullrequestreview-1538769707

Reply:
sooo we can get rid of it /eventually/, but it s [a bit more involved than I'd like](https://github.com/apollographql/router/pull/3505). maybe if we followup